commit a6e18b27b06bf1ecc42800ab6ad622fc5bd2b50e
Author: Nathan Chancellor <natechancellor@gmail.com>
Date:   Sun Mar 4 17:20:25 2018 -0700

    sound: bcm: Fix memset dereference warning
    
    This warning appears with GCC 6.4.0 from toolchains.bootlin.com:
    
    ../sound/soc/bcm/allo-piano-dac-plus.c: In function ‘snd_allo_piano_dac_init’:
    ../sound/soc/bcm/allo-piano-dac-plus.c:711:30: warning: argument to ‘sizeof’ in ‘memset’ call is the same expression as the destination; did you mean to dereference it? [-Wsizeof-pointer-memaccess]
      memset(glb_ptr, 0x00, sizeof(glb_ptr));
                                  ^
    
    Suggested-by: Phil Elwell <phil@raspberrypi.org>
    Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>

diff --git a/sound/soc/bcm/allo-piano-dac-plus.c b/sound/soc/bcm/allo-piano-dac-plus.c
index d35e0553a763..35d77e905fdf 100644
--- a/sound/soc/bcm/allo-piano-dac-plus.c
+++ b/sound/soc/bcm/allo-piano-dac-plus.c
@@ -706,11 +706,10 @@ static int snd_allo_piano_dac_init(struct snd_soc_pcm_runtime *rtd)
 	struct snd_soc_card *card = rtd->card;
 	struct glb_pool *glb_ptr;
 
-	glb_ptr = kmalloc(sizeof(struct glb_pool), GFP_KERNEL);
+	glb_ptr = kzalloc(sizeof(struct glb_pool), GFP_KERNEL);
 	if (!glb_ptr)
 		return -ENOMEM;
 
-	memset(glb_ptr, 0x00, sizeof(glb_ptr));
 	card->drvdata = glb_ptr;
 	glb_ptr->dual_mode = 2;
 	glb_ptr->set_mode = 0;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1595_a6e18b27b06bf1ecc42800ab6ad622fc5bd2b50e.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1595_a6e18b27b06bf1ecc42800ab6ad622fc5bd2b50e.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
